sdks/fabric/interoperation-node-sdk/src/InteroperableHelper.ts

```diff
@@ -170,19 +171,64 @@ const verifyRemoteProposalResponse = async (proposalResponseBase64, isEncrypted,
-        if (interopPayload.getConfidential()) {    // Currently this is only supported for Fabric because it uses ECDSA keys in wallets
-            const confidentialPayload = interopPayloadPb.ConfidentialPayload.deserializeBinary(Uint8Array.from(Buffer.from(interopPayload.getPayload())));
-            const decryptedPayload = decryptData(Buffer.from(confidentialPayload.getEncryptedPayload()), privKeyPEM);
-            const decryptedPayloadContents = interopPayloadPb.ConfidentialPayloadContents.deserializeBinary(Uint8Array.from(Buffer.from(decryptedPayload)));
-            return { interopPayload: interopPayload, data: Buffer.from(decryptedPayloadContents.getPayload()).toString(), contents: decryptedPayload };
+        const fabricViewProposalResponses = fabricView.getEndorsedProposalResponsesList();
+        let viewAddress = '';
+        let responsePayload = '';
+        let responsePayloadContents = null;
+        let paymentConfidential = false;
+        for (let i = 0; i < fabricViewProposalResponses.length; i++) {
+            const fabricViewChaincodeAction = proposalPb.ChaincodeAction.deserializeBinary(fabricViewProposalResponses[i].getPayload().getExtension_asU8());
+            const interopPayload = interopPayloadPb.InteropPayload.deserializeBinary(Uint8Array.from(Buffer.from(fabricViewChaincodeAction.getResponse().getPayload())));
+            if (interopPayload.getConfidential()) {    // Currently this is only supported for Fabric because it uses ECDSA keys in wallets
+                const confidentialPayload = interopPayloadPb.ConfidentialPayload.deserializeBinary(Uint8Array.from(Buffer.from(interopPayload.getPayload())));
+                const decryptedPayload = decryptData(Buffer.from(confidentialPayload.getEncryptedPayload()), privKeyPEM);
+                const decryptedPayloadContents = interopPayloadPb.ConfidentialPayloadContents.deserializeBinary(Uint8Array.from(Buffer.from(decryptedPayload)));
+                if (i === 0) {
+                    viewAddress = interopPayload.getAddress();
+                    responsePayload = Buffer.from(decryptedPayloadContents.getPayload()).toString();
+                    responsePayloadContents = decryptedPayload;
+                    paymentConfidential = true;
+                } else if (!paymentConfidential) {
+                    throw new Error(`Mismatching payment confidentiality flags across proposal responses`);
+                } else {
+                    const currentResponsePayload = Buffer.from(decryptedPayloadContents.getPayload()).toString();
+                    if (responsePayload !== currentResponsePayload) {
+                        throw new Error(`Decrypted proposal response payloads mismatch: 0 - ${responsePayload}, ${i} - ${currentResponsePayload}`);
+                    }
+                    if (!responsePayloadContents.equals(decryptedPayload)) {
```

Comment:
`responsePayloadContents` contains the actual payload and the random bytes right?
will `responsePayloadContents` match over different peers? given that every peer is generating its own random bytes in the payload?

Reply:
The decrypted payloads ought to match, otherwise the contract code itself must be non-deterministic, which poses safety concerns of the kinds that our system can't catch (that's something the smart contract developers need to ensure doesn't happen.)

Reply:
You are right, good catch! This matching logic is unnecessary, both in the SDK and in the chaincode. All we need to do is ensure that each hash in the array is consistent with the supplied plaintext.